core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+        if item.sent_datetime > current_datetime:
+            cls.errors['sent datetime check'] = (
+                'Model id %s: The sent_datetime field has a value %s which is '
+                'greater than the time when the job was run'
+                ) % (item.id, item.sent_datetime)
+
+    @classmethod
+    def _validate_sender_email(cls, item):
+        """Validate that sender email corresponds to email of user obtained
+        by using the sender_id.
+
+        Args:
+            item: SentEmailModel to validate.
+        """
+        _, sender_models = (cls.external_models['sender_id'])
+        sender_model = sender_models[0][1]
+        if sender_model and not sender_model.deleted:
+            if sender_model.email != item.sender_email:
+                cls.errors['sender email check'] = (
+                    'Model id %s: Sender email %s in model does not match '
+                    'with email %s of user obtained through sender id') % (
+                        item.id, item.sender_email, sender_model.email)
+
+    @classmethod
+    def _validate_recipient_email(cls, item):
+        """Validate that recipient email corresponds to email of user obtained
+        by using the recipient_id.
+
+        Args:
+            item: SentEmailModel to validate.
+        """
+        _, recipient_models = (cls.external_models['recipient_id'])
+        recipient_model = recipient_models[0][1]
+        if recipient_model and not recipient_model.deleted:
+            if recipient_model.email != item.recipient_email:
+                cls.errors['recipient email check'] = (
+                    'Model id %s: Recipient email %s in model does not match '
+                    'with email %s of user obtained through recipient id') % (
+                        item.id, item.recipient_email, recipient_model.email)
+
+    @classmethod
+    def _get_validation_functions(cls):
+        return [
+            cls._validate_sent_datetime, cls._validate_sender_email,
+            cls._validate_recipient_email]
+
+
+class BulkEmailModelValidator(BaseModelValidator):
+    """Class for validating BulkEmailModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'recipient_id': (
+                user_models.UserSettingsModel, item.recipient_ids),
+            'sender_id': (user_models.UserSettingsModel, [item.sender_id]),
+        }
+
+    @classmethod
+    def _validate_id_length(cls, item):
+        """Validate that model id has length 12.
+
+        Args:
+            item: BulkEmailModel to validate.
+        """
+        if len(item.id) != 12:
```

Comment:
Add a comment explaining this check. Instead of 12 can we use some constant here?

Reply:
Done.